crf/src/main/java/org/sagebase/crf/step/active/BpmRecorder.java

```diff
@@ -213,10 +217,15 @@ public HeartBeatJsonWriter(BpmUpdateListener
 
         private int sampleCount = 0;
         private double timestampReference = -1;
+        private double firstTimeStamp = -1;
 
         @AnyThread
         @Override
         public void onHeartRateSampleDetected(HeartBeatSample sample) {
+            if (firstTimeStamp == -1) {
+                firstTimeStamp = sample.timestamp;
```

Comment:
Strictly speaking, this isn't when the video starts but instead when the step transitions from "stop, your three minute step test is done" to "now measure your heart rate". It's not a big difference but at least on iOS devices, starting up the video can take a few seconds.